RecoVertex/PrimaryVertexProducer/src/AdaptiveChisquarePrimaryVertexFitter.cc

```diff
@@ -0,0 +1,540 @@
+    return SBeam;
+  } else {
+    edm::LogWarning("AdaptiveChisquarePrimaryVertexFitter")
+        << "Warning, beam-spot covariance matrix inversion failed " << std::endl;
+    Error3 S0;
+    S0(0, 0) = 1. / pow(beamspot.BeamWidthX(), 2);
+    S0(1, 1) = 1. / pow(beamspot.BeamWidthY(), 2);
+    S0(2, 2) = 1. / pow(beamspot.sigmaZ(), 2);
+    return S0;
+  }
+}
+
+TransientVertex AdaptiveChisquarePrimaryVertexFitter::get_TransientVertex(
+    const unsigned int k,
+    const std::vector<std::pair<unsigned int, float>> &vertex_track_weights,
+    const std::vector<reco::TransientTrack> &tracks,
+    const float beam_weight,
+    const reco::BeamSpot &beamspot) {
+  const GlobalPoint pos(xv[k], yv[k], zv[k]);
+  const GlobalError posError(
+      V_vtx[k](0, 0), V_vtx[k](1, 0), V_vtx[k](1, 1), V_vtx[k](2, 0), V_vtx[k](2, 1), V_vtx[k](2, 2));
+  float chi2 = 0.;
+  float vtx_ndof = -3.;
+  if (beam_weight > 0) {
+    // add beam-spot chi**2 and degrees of freedom
+    vtx_ndof = 3 * beam_weight;
+    const auto S = get_inverse_beam_covariance(beamspot);
+    const double dx = xv[k] - beamspot.x0();
+    const double dy = yv[k] - beamspot.y0();
+    const double dz = zv[k] - beamspot.z0();
+    chi2 = beam_weight * (S(0, 0) * dx * dx + S(1, 1) * dy * dy + 2 * S(0, 1) * dx * dy + S(2, 2) * dz * dz +
+                          2 * S(0, 2) * dx * dz + 2 * S(1, 2) * dy * dz);
+  }
+
+  std::vector<reco::TransientTrack> vertex_tracks;
+  TransientVertex::TransientTrackToFloatMap trkWeightMap;
+  for (const auto &tk : vertex_track_weights) {
+    const unsigned int i = tk.first;
+    const float track_weight = tk.second;
+    if (track_weight >= min_trackweight_) {
+      vertex_tracks.emplace_back(tracks[i]);
+      trkWeightMap[tracks[i]] = track_weight;
+      vtx_ndof += 2 * track_weight;
+      chi2 += track_weight * track_in_vertex_chsq(trackinfo[i], xv[k], yv[k], zv[k]);
+    }
+  }
+
+  auto vtx = TransientVertex(pos, posError, vertex_tracks, chi2, vtx_ndof);
+  vtx.weightMap(trkWeightMap);
+
+  return vtx;
+}
+
+std::vector<TransientVertex> AdaptiveChisquarePrimaryVertexFitter::vertices(
+    const std::vector<reco::TransientTrack> &tracks,
+    const std::vector<TransientVertex> &clusters,
+    const reco::BeamSpot &beamspot,
+    const bool useBeamConstraint) {
+  // simultaneous fit of all vertices in the input list
+
+  const int max_iterations = 50;
+
+  // initialize the vertices
+  const unsigned int nv = clusters.size();
+  xv.clear();
+  xv.reserve(nv);
+  yv.clear();
+  yv.reserve(nv);
+  zv.clear();
+  zv.reserve(nv);
+  tkfirstv.clear();
+  tkfirstv.reserve(nv + 1);
+  V_vtx.clear();
+  V_vtx.reserve(nv);
+
+  // seeds
+  for (auto &clu : clusters) {
+    const double zclu = clu.position().z();
+    xv.emplace_back(beamspot.x(zclu));
+    yv.emplace_back(beamspot.y(zclu));
+    zv.emplace_back(zclu);
+  }
+
+  fill_trackinfo(tracks, beamspot);
+
+  const double beta = 1.0;  // no annealing
+  make_vtx_trk_map(5 * sqrt(beta));
```

Comment:
Is this the only place beta is used?
While it's nice to explain that this corresponds to the no annealing case, perhaps we could just remove the sqrt(1.0) evaluation.

Reply:
indeed, at this point it is a remnant of an unused feature and should be removed here.